### ISSUE TITLE
Pixel build failures

### ISSUE THREAD

**Issue by Rudd-O** [copperhead/bugtracker#737]
```
+ build/tools/releasetools/sign_target_files_apks -o -d keys/marlin --replace_verity_public_key keys/marlin/verity_key.pub --replace_verity_private_key keys/marlin/verity --replace_verity_keyid keys/marlin/verity.x509.pem out/target/product/marlin/obj/PACKAGING/target_files_intermediates/aosp_marlin-target_files-2017.09.10.17.zip out/release-marlin-2017.09.10.17/marlin-target_files-2017.09.10.17.zip
ERROR: no key specified for:

  webview.apk

Use '-e <apkname>=' to specify a key (which may be an
empty string to not sign this apk).
+ exit 1
[Pipeline] }
[Pipeline] // stage
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
ERROR: script returned exit code 1
Finished: FAILURE
```

**Comment by thestinger** [comment]
`rm -r out` should fix it. You should do that before production releases in general, since it won't handle all changes properly via incremental builds.

**Comment by thestinger** [comment]
(or `make clean` / `make clobber` but I prefer `rm -r out` because it's quicker and fully clears away everything it built, including stuff related to build system itself)

**Comment by Rudd-O** [comment]
OK.  Please add that to the docs.  Thank you.  Note that doing that will cause the build to take six hours, instead of 1 hour.

**Comment by Rudd-O** [comment]
Note that the solution does not make me happy at all because of the increased build times. ninja should not have a problem dealing with incremental builds.

**Comment by thestinger** [comment]
I did add it to the documentation recently.

> ninja should not have a problem dealing with incremental builds.

Incremental builds usually work but that's not guaranteed and they often don't pick up all changes completely. Removed files / modules don't get removed from the generated code. For development, it possible to get away with doing stuff like `rm -r out/target/product/sailfish/` to deal with issues like WebView being removed but that isn't a reliable way of handling all changes. Proper production builds really need to be clean builds.

Similarly, changes to the toolchain are not always picked up, especially across the board. The only safe, complete option for production is a clean build.
